src/lists.h

```diff
@@ -862,17 +863,16 @@ extern UInt SetFiltListTNums [ LAST_REAL_TNUM ] [ LAST_FN + 1 ];
 **
 *F  SET_FILT_LIST( <list>, <fnum> ) . . . . . . . . . . . . . .  set a filter
 */
-#define SET_FILT_LIST(list,fn) \
-  do { \
-    UInt     n; \
-    n = SetFiltListTNums[TNUM_OBJ(list)][fn]; \
-    if ( n != (UInt)-1 ) \
-      RetypeBagIfWritable( list, n ); \
-     else { \
-      Pr( "#E  SET_FILT_LIST[%s][%d] in ", (Int)TNAM_OBJ(list), fn ); \
-      Pr( "%s:%d\n", (Int)__FILE__, (Int)__LINE__); \
-      }  \
-  } while (0)
+static inline void SET_FILT_LIST(Obj list, EnumFnIs fn)
+{
+    UInt n = SetFiltListTNums[TNUM_OBJ(list)][fn];
+    GAP_ASSERT(n != 0);
```

Comment:
How about replacing the `GAP_ASSERT` with his: `if (n == 0) return;`.  Then you are free to call `SET_FILT_LIST` on arbitrary objects:

Reply:
I'm not 100% clear on what all this code does, but there are clearly some objects we aren't supposed to call this function on, the ones where `n = (UInt)-1`, as we print a mandatory warning message in those cases. Therefore I was worried about starting to accept values which shouldn't be passed in.

Reply:
did it anyway